src/components/plotly/pyramid-bar/index.js

```diff
@@ -0,0 +1,104 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { plotlyDefaultProps, plotlyPropTypes } from '../shared/constants'
+import CustomPlot from '../shared/custom-plot'
+import useTransformedData from '../shared/use-transformed-data'
+
+
+const PyramidBar = ({
+  data,
+  stacked,
+  showTicks,
+  showAxisTitles,
+  x,
+  y,
+  orientation,
+  percentage,
+  ...props
+}) => {
+  const getSum = () => {
+    let total = 0
+    y.forEach(k => {
+      const sum = data.map(d => d[k]).reduce((acc, val) => acc + val, 0)
+      total += sum
+    })
+
+    return total
+  }
+
+
+  const _data = useTransformedData({
+    type: 'bar',
+    data,
+    y,
+    orientation,
+    variant: 'pyramidBar',
+    percentage,
+    sum: getSum(),
+    ...props,
+  })
+
+  const xValReverse = x.slice().reverse()
+  const maxVal = Math.max(...x)
+
+  const getTickText = () => {
+    let tickText = [...x, 0, ...xValReverse]
+
+    if (percentage) { 
+      const transformPercentage = x.map(val => `${Math.round((val*100) / getSum())}%`)
+      tickText = [...transformPercentage, '0%', ...transformPercentage.slice().reverse()]
+    }
+
+    return tickText
+  }
+
+
+  return (
+    <CustomPlot
+      type='bar'
+      data={_data}
+      layout={{
+        barmode: 'relative',
+        bargap: 0.1,
+        hovermode: false,
+        xaxis: {
```

Comment:
can we allow optional (similar to y-axis) prop to display a x-axis label? such that one can use the component as:

```jsx
<PyramidBar
  ...
  xAxisLabel='count'
/>
```

and it would be displayed as the x-axis label (perhaps on the right-hand-side)